deps/rabbitmq_auth_backend_ldap/src/rabbit_auth_backend_ldap_mgmt.erl

```diff
@@ -0,0 +1,196 @@
+        UseStartTls = rabbit_mgmt_util:parse_bool(maps:get(use_starttls, BodyMap, false)),
+        Servers = maps:get(servers, BodyMap, []),
+        UserDN = maps:get(user_dn, BodyMap, <<"">>),
+        Password = maps:get(password, BodyMap, <<"">>),
+        Options0 = [
+            {port, Port},
+            {timeout, 5000}
+        ],
+        try
+            {ok, Options1} = maybe_add_ssl_options(Options0, UseSsl, BodyMap),
+            case eldap:open(Servers, Options1) of
+                {ok, LDAP} ->
+                    Result = case maybe_starttls(LDAP, UseStartTls, BodyMap) of
+                        ok ->
+                            case eldap:simple_bind(LDAP, UserDN, Password) of
+                                ok ->
+                                    {true, ReqData1, Context};
+                                {error, invalidCredentials} ->
+                                    rabbit_mgmt_util:not_authorised("invalid credentials", ReqData1, Context);
+                                {error, unwillingToPerform} ->
+                                    rabbit_mgmt_util:not_authorised("invalid credentials", ReqData1, Context);
+                                {error, E} ->
+                                    Reason = unicode_format(E),
+                                    rabbit_mgmt_util:bad_request(Reason, ReqData1, Context)
+                            end;
+                        Error ->
+                            Reason = unicode_format(Error),
+                            rabbit_mgmt_util:bad_request(Reason, ReqData1, Context)
+                    end,
+                    eldap:close(LDAP),
+                    Result;
+                {error, E} ->
+                    Reason = unicode_format(E),
+                    rabbit_mgmt_util:bad_request(Reason, ReqData1, Context)
+            end
+        catch throw:{bad_request, ErrMsg} ->
+            rabbit_mgmt_util:bad_request(ErrMsg, ReqData1, Context)
+        end
+    end,
+    rabbit_mgmt_util:with_decode([], ReqData0, Context, F).
+
+%%--------------------------------------------------------------------
+
+unicode_format(Arg) ->
+    rabbit_data_coercion:to_utf8_binary(io_lib:format("~tp", [Arg])).
+
+maybe_starttls(_LDAP, false, _BodyMap) ->
+    ok;
+maybe_starttls(LDAP, true, BodyMap) ->
+    {ok, TlsOptions} = tls_options(BodyMap),
+    eldap:start_tls(LDAP, TlsOptions, 5000).
+
+maybe_add_ssl_options(Options0, false, _BodyMap) ->
+    {ok, Options0};
+maybe_add_ssl_options(Options0, true, BodyMap) ->
+    case maps:is_key(ssl_options, BodyMap) of
+        false ->
+            {ok, Options0};
+        true ->
+            Options1 = [{ssl, true} | Options0],
+            {ok, TlsOptions} = tls_options(BodyMap),
+            Options2 = [{sslopts, TlsOptions} | Options1],
+            {ok, Options2}
+    end.
+
+tls_options(BodyMap) ->
+    case maps:get(ssl_options, BodyMap, undefined) of
```

Comment:
minor style nit: we could match on `ssl_options` being set in `BodyMap` to get rid of this `case` and the extra indentation it makes